compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionPreprocessing.cpp

```diff
@@ -69,6 +69,31 @@ struct GenericOpInterchangePattern
   }
 };
 
+//===----------------------------------------------------------------------===//
+// ElementwiseOpInterchangePattern
+//===----------------------------------------------------------------------===//
+
+struct ElementwiseOpInterchangePattern
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(linalg::GenericOp genericOp,
+                                PatternRewriter &rewriter) const override {
+    if (!linalg::isElementwise(genericOp) || genericOp.getNumResults() != 1)
```

Comment:
Does this also work if the operation has "reduction" iterator type. Please add a test for that if it does. Thanks!

Reply:
I don't think reduction would work because the goal is to have identity output maps but a reduction iterator would the output map would have to be a projection